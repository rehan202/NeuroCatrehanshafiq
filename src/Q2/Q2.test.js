import React from 'react';
import ReactDOM from 'react-dom';
import Button from './q2';
import { render, cleanup } from '@testing-library/react';

afterEach(cleanup);

it("renders without Crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button></Button>, div);
})

it("button has name prop and rendering correctly", () => {
    const { getByTestId } = render(<Button name="Click me please"></Button>)
    expect(getByTestId('myButton')).toHaveTextContent("Click me please")
})

it("renders button snapshot test", () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
});
