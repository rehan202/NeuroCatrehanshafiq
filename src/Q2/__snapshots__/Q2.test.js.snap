// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button snapshot test 1`] = `
<DocumentFragment>
  <div>
    <button
      data-testid="myButton"
      style="display: inline; background-color: Green;"
    />
  </div>
</DocumentFragment>
`;
